### PR TITLE
sg: Add `sg rfc [list|search|open]` commands

### PR DESCRIPTION
**Pull request by mrnugget**
Demo video:

https://user-images.githubusercontent.com/1185253/125973447-ee13e2be-b644-44ae-a910-7225cf13403b.mp4

### CODE REVIEW

**Review by jeanduplessis — APPROVED** [review]

File: dev/sg/rfc.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+			out.Writef("RFC %s%s %s%s%s %s", numberColor, number, statusColor, status, output.StyleReset, name)
+		} else {
+			out.Writef("%s%s", i.Name, output.StyleReset)
+		}
+	}
+
+	return nil
+}
+
+func openURL(url string) error {
```
Comment (jeanduplessis) [inline dev/sg/rfc.go:170]: We should consider extracting this into a utils package at some point

File: dev/sg/rfc.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	list := srv.Files.List().
+		Corpora("drive").SupportsAllDrives(true).
+		DriveId("0AK4DcztHds_pUk9PVA").
+		IncludeItemsFromAllDrives(true).
+		SupportsAllDrives(true).
+		PageSize(100).
+		Q(q).
+		Fields("nextPageToken, files(id, name, parents)")
+
+	if orderBy != "" {
```
Comment (jeanduplessis) [inline dev/sg/rfc.go:126]: FYI: When querying over the fullText field, as in the `searchRFCs` function, you can't use orderBy which is why I added this check.

**Comment by mrnugget** [comment]
@jeanduplessis I fixed the issue of `sg rfc [...]` not working outside `sourcegraph/sourcegraph`. I removed the implicit call to the `root.RepositoryRoot` function in `internal/db/db.go` and added explicit calls to `parseConf` to all commands that need it. That's a bit of boilerplate that we will probably need to clean up in the future. But it's fine for now.

**Comment by mrnugget** [comment]
@tjdevries tagging you just for an FYI that I touched your `internal/db/db.go` code a bit.

File: dev/sg/rfc.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	"github.com/sourcegraph/sourcegraph/lib/output"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/drive/v3"
+	"google.golang.org/api/option"
+)
+
+const (
+	tokenFileName = ".sg.token.json"
+	credentials   = `{"installed":{"client_id":"1043390970557-1okrt0mo0qt2ogn2mkp217cfrirr1rfd.apps.googleusercontent.com","project_id":"sg-cli","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"gkQ2alKQZr2088IFGr55ET_I","redirect_uris":["urn:ietf:wg:oauth:2.0:oob","http://localhost"]}}`
```
Comment (tjdevries) [inline dev/sg/rfc.go:25]: what is this?

**Comment by tjdevries** [comment]
db changes seem fine, thanks :)

File: dev/sg/rfc.go (hunk 3 above)
Comment (mrnugget) [reply dev/sg/rfc.go:25]: Static oauth2 client data. See [this query](https://sourcegraph.com/search?q=context:global+%22token_uri%22:%22https://oauth2.googleapis.com/token%22&patternType=literal) :smile:

**Pull request merged**